### PR TITLE
Add support for python 3.10

### PR DESCRIPTION
**Pull request by gherceg**
https://dimagi-dev.atlassian.net/browse/SAAS-13500

First step in upgrading commcare-cloud to work with python 3.10. Once tests pass, we should be able to test this on staging.

##### ENVIRONMENTS AFFECTED

None

### CODE REVIEW

File: src/commcare_cloud/parse_help.py
Comment (millerdev) [inline src/commcare_cloud/parse_help.py:19]: nit
```suggestion
    assert sys.version_info.major == 3, sys.version_info
```

**Comment by gherceg** [comment]
There are two entry points in our build that depend on the generated help files from the `ansible -h` and `ansible-playbook -h` commands. As of 3.10, the output for `<ansible-command> -h` changed ever so slightly  from "optional arguments:" to "options:". This is important because we use [this syntax](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/parse_help.py#L10) to key off when parsing help messages ([here](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/commands/ansible/run_module.py#L66) and [here](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/commands/ansible/ansible_playbook.py#L62))

The first entrypoint occurs when running `commcare-cloud testenv deploy-stack --skip-check --quiet --tags=py3,commcarehq`. In this case, the "cached" files are used as is from the repository which at the moment means they were generated with python 3.6 and have "optional arguments:" syntax.

The second entrypoint occurs when running the `test_autogen_docs.sh` script, which starts by running `make clean` and removing all "cached" files, and then regenerating them. This means that when running with 3.10, the files contain the new "options:" syntax.

The logic to resolve the parsing is straightforward, and could either be the current implementation which is to grab the correct syntax based on the python version, or just use a regex pattern to match against that contains both strings (leaning towards changing to this).

I think the more important factor is how to handle the git diff that results from generating these files on an env with 3.10. The change to the ansible.txt and ansible-playbook.txt files on 3.10 are flagged by the `fail_if_there_is_a_git_diff.sh` script as not being logged properly in our changelog. This would make it tricky to support multiple versions of python currently (assuming that is desirable), so we would either:
- remove these files from git and leave it up to specific environments to generate them
- continue to only target one version of python at a time

**Comment by millerdev** [comment]
Could we do something to eliminate the diffs in those files? Like post-process the ansible help output to standardize lines with expected diffs before saving it in the "cache" file?

File: src/commcare_cloud/parse_help.py
Comment (gherceg) [inline src/commcare_cloud/parse_help.py:23]: I added this when playing around with removing the cache entirely as I noticed this returned a byte string previously. At the moment it looks like we never actually invoke this code since only the two ansible commands use this method, and definitely hit the cache. I think this is a safe change, but wanted to flag since it isn't directly related to any 3.10 compatibility.

File: setup.py
Comment (dannyroberts) [inline setup.py:35]: Looks like `nose` hasn't received an update since 2015. Is it time to remove it from our test stack here? (Not blocking this PR.)

**Review by dannyroberts — APPROVED** [review]

File: Makefile
Comment (millerdev) [inline Makefile:12]: Would it be disruptive to do the opposite replacement, ending up with `options:`, since that is the format supported by newer versions like Python 3.10? That way this could be removed at some point when we drop support for older versions of Python that output `optional arguments:`.

File: setup.py
Comment (millerdev) [reply setup.py:35]: This is a bigger topic, so I agree that it should not block this PR. My vote is to keep nose until pytest has a solution for https://github.com/pytest-dev/pytest/issues/3834

**Review by millerdev — APPROVED** [review]

File: Makefile
Comment (gherceg) [reply Makefile:12]: That does make more sense. Will update shortly.

Comment (gherceg) [reply Makefile]: See 88f370a999a0d720f5bf99ca35ea980a71d9fe52

**Review by millerdev — APPROVED** [review]

**Pull request merged**
